src/mono/mono/metadata/mono-mlist.c

```diff
@@ -69,7 +69,7 @@ mono_mlist_alloc_checked (MonoObject *data, MonoError *error)
 	MonoMList* res;
 	if (!monolist_item_vtable) {
 		MonoClass *klass = mono_class_load_from_name (mono_defaults.corlib, "Mono", "MonoListItem");
-		monolist_item_vtable = mono_class_vtable_checked (mono_get_root_domain (), klass, error);
+		monolist_item_vtable = mono_class_vtable_checked (klass, error);
```

Comment:
mono-mlist.[ch] aren't in the CMakeLists.txt anymore. I think the files can be removed